src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs

```diff
@@ -658,6 +659,18 @@ protected override void RegisterCallbacks()
                 FramebufferResize?.Invoke(new(width, height));
             };
 
+            _onRefresh = (window) =>
+            {
+                if (!IsClosing)
+                {
+                    DoUpdate();
+                }
+                if (!IsClosing)
+                {
+                    DoRender();
+                }
+            };
```

Comment:
This is fine, and I'm pretty happy to just merge as is, but one possible Q which I'm not sure how much we should care about is "what if they're not using our Render/Update events?"

What we could do:
- Have a boolean field on the window e.g. `_isInDoEvents` that is set to `true` when `DoEvents` is called.
- If `DoEvents` is called and this is already `true`, `DoEvents` does not execute
- In addition to the `Run` implementation just being a `while (true)` loop by default, we could capture the `Action` passed in and use that for `_onRefresh` instead.

The reason why we need the boolean flag is because the user's action will naturally try to call `DoEvents`, which would cause infinite recursion/undefined behaviour.

So sorry I didn't think of this earlier, I really don't want to mess you around with this too much. This is technically fine so we can merge this as is if you want, but I just remembered that this is exactly the sort of scenario that `Run(Action)` was designed for!

Reply:
Can you please give me a test case for such an Update() event?  I am not 100% sure I follow and I want to make sure I cover your concern.